src/Sulu/Component/Content/Types/ResourceLocator/Strategy/TreeTransliteratorGeneratorDecorator.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Types\ResourceLocator\Strategy;
+
+/**
+ * Uses Intl component to transliterate generated titles.
+ */
+class TreeTransliteratorGeneratorDecorator implements ResourceLocatorGeneratorInterface
+{
+    /** @var ResourceLocatorGeneratorInterface */
+    private $originalGenerator;
+
+    /** @var \Transliterator */
+    private $transliterator;
+
+    public function __construct(ResourceLocatorGeneratorInterface $originalGenerator, \Transliterator $transliterator)
+    {
+        $this->originalGenerator = $originalGenerator;
+        $this->transliterator = $transliterator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate($title, $parentPath = null)
+    {
+        $generated = $this->originalGenerator->generate($title, $parentPath);
+
+        return $this->transliterator->transliterate($generated);
```

Comment:
Where do you configure the language for the `Transliterator`? I.e. what you have set to "Russian-Latin/BGN" in the test?

Reply:
@danrot you pass already configured Transliterator to Decorator. You can see the docs PR.